examples/dataflow-data-generator/data-generator-pipeline/data_generator/ParquetUtil.py

```diff
@@ -0,0 +1,42 @@
+import pyarrow as pa
+import logging
+import json
+
+
+def get_pyarrow_translated_schema(string_schema):
+    type_conversions = {
+        'STRING': pa.string(),
+        'NUMERIC': pa.int64(),
+        'BYTE': None,
+        'INTEGER': pa.int64(),
+        'FLOAT': pa.float64(),
+        'NUMERIC': pa.int64(),
+        'BOOLEAN': pa.bool_(),
+        'TIMESTAMP': pa.timestamp('ms'),
+        'DATE': pa.date32(),
+        'TIME': pa.time64('us'),
+        'DATETIME': pa.date64(),
+        'GEOGRAPHY': None,
+        'RECORD': None
+    }
+    pa_schema_list = []
+    for field in string_schema:
+        field_type = field['type']
+        field_name = field['name']
+        converted_field_type = type_conversions[field_type]
+        if converted_field_type is None:
+            logging.error('Error: json schema included a {0:s} field. BYTE,  '
+                          'GEOGRAPHY, and RECORD types cannot be used when '
+                          'outputting to parquet.'.format(field_type))
+            raise ValueError('Error: json schema included a {0:s} field. BYTE, '
+                             'GEOGRAPHY, and RECORD types cannot be used when '
+                             'outputting to parquet.'.format(field_type))
+        else:
+            pa_field = pa.field(
+                field_name,
+                converted_field_type
```

Comment:
you need to check if the field is NULLABLE in the `string_schema` and set nullability with the kwargs `nullable=True` or `nullable=False`in the constructor for `pa_field`

Reply:
Additionally, you should handle creating an array if the input field mode is REPEATED.
You'll have to do this using the [`pyarrow.list_(value_type)`](https://arrow.apache.org/docs/python/generated/pyarrow.list_.html#pyarrow.list_) factory method.